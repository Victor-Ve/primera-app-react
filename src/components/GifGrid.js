import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({ categoria }) => {
  const { data: imagenes, loading } = useFetchGifs(categoria);

  return (
    <>
      <h3 className="card animate__animated animate__zoomInDown">
        {categoria}
      </h3>
      {loading && <p>Loading</p>}
      {
        <div className="card-grid">
          {imagenes.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </div>
      }
    </>
  );
};

GifGrid.propTypes = {
  categoria: PropTypes.string.isRequired,
};

export default GifGrid;
