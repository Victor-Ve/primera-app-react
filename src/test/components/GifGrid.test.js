import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {
  const categoria = 'hola';
  test('Snapshots del componente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid categoria={categoria} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar item cuando se cargan imagenes useFectchGifs', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://localhost/cualqueir/cosa.jpg',
        title: 'cualqueir cosa',
      },
      {
        id: '123',
        url: 'https://localhost/cualqueir/cosa.jpg',
        title: 'cualqueir cosa',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid categoria={categoria} />);

    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    expect(wrapper.find('p').exists()).toBe(false);
    // expect(wrapper).toMatchSnapshot(); poco eficaz
  });
});
