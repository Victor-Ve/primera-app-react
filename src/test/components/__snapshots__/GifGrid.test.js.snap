// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Snapshots del componente 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__zoomInDown"
  >
    hola
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
